src/GazeboThruster.cpp

```diff
@@ -175,7 +192,25 @@ void GazeboThruster::updateBegin(common::UpdateInfo const& info)
             thruster != thrusters.end(); ++thruster)
     {
         physics::LinkPtr link = model->GetLink( thruster->name );
-        link->AddRelativeForce( math::Vector3(thruster->effort,0,0) );
+        link->AddLinkForce( thruster->effort*gazebo::math::Vector3::UnitX);
+        link->AddLinkForce( thruster->effort*thruster->direction, thruster->position );
     }
 }
 
+void GazeboThruster::updateCompensatedEffort(gazebo_underwater::Matrix6 const& matrix, gazebo::math::Vector3 const& cog)
+{
+    gazebo_underwater::Vector6 effort;
```

Comment:
This should be defined when it's used. Defining it here suggests that its value is kept across loops - which it is not.